### PR TITLE
Support for user defined metadata in 'album.ini' file

### PR DESCRIPTION
**Pull request by pmetras**
These changes add support for a configuration file name `album.ini` that the user can place into photo albums to define user defined metadata. This is useful when media don't have some metadata (for instance geotagging) or when the metadata extracted from the media must be changed (for example when geoname does not find the correct location). By metadata, I mean media all type of sidecar information, not only metadata from EXIF part of photo files. More explanations in #61.

I've added a documentation file that explains how to use that feature.

This version only implement the bases of custom metadata, in the scanner code. The changes to the web application are limited: displaying only 3 new metadata. The web application could be enhanced by using the `tags` metadata to allow new ways to browse or search media. As it said in school book, "this is let to the reader as an exercise" for the moment.

**Known bugs or limitations:**
* The scanner does not detect when `album.ini` files are changed. The user has to force a complete scan to take into account the changes.
* Adapt the web application to use new metadata.

### CODE REVIEW

**Comment by paolobenve** [comment]
good work! you only missed to pull some commit from master

**Comment by pmetras** [comment]
Sorry. I'm still learning git workflows. I hope it's better now...

**Comment by paolobenve** [comment]
cannot understand why... your scanner/cachePath.py is different from mine...

**Comment by paolobenve** [comment]
supposedly you pulled my master branch to your master branch, you must merge your master branch to you album_ini branch too

**Comment by pmetras** [comment]
Well, the CachePath.py was not committed on my computer. Now I've pushed it, but I'm still not sure of the result. I'll check later and try to merge master.

**Comment by pmetras** [comment]
OK, now if I've done the job correctly, it should be good. My problem is that I'm working on two computers and I messed with the changes in the files.

File: doc/Metadata.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+# These default values apply to all media content in the directory
+[DEFAULT]
+# Manual geotagging of all pictures
+# You can use https://www.latlong.net/, https://www.gps-coordinates.org/ or similar service to find places
+# All photos and videos in the current album, including the album itself, will appear to have
+# been takes in Montreal, Canada
+longitude = -73.588014
+latitude = 45.508873
+
+# When album metadata is developped, this information will be used. For the moment, it's useless
```
Comment (paolobenve) [inline doc/Metadata.md]: well, albums already have a date-time metadata, it's the most recent date-time of the included media and subalbums

**Pull request merged**